nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/LoginCommand.java

```diff
@@ -120,21 +121,32 @@ protected int executeCommand() throws CommandException, CommandValidationExcepti
      * Prompt for the admin user name.
      */
     private String getAdminUser() {
-        Console cons = System.console();
         String user = null;
-        String defuser = programOpts.getUser();
-        if (defuser == null) {
-            defuser = SystemPropertyConstants.DEFAULT_ADMIN_USER;
-        }
-        if (cons != null) {
-            cons.printf("%s", strings.get("AdminUserPrompt", defuser));
-            String val = cons.readLine();
-            if (val != null && val.length() > 0){
-                user = val;
-            } else {
-                user = defuser;
+
+        try (ConsoleReader cons = new ConsoleReader(System.in, System.out, null)) {
+            String defuser = programOpts.getUser();
+            if (defuser == null) {
+                defuser = SystemPropertyConstants.DEFAULT_ADMIN_USER;
             }
+            if (cons != null) {
+                cons.setPrompt(strings.get("AdminUserPrompt", defuser));
+
+                String val = null;
+                try {
+                    val = cons.readLine();
+                } catch (IOException ioe) {
+                    logger.log(Level.WARNING, "Error reading input", ioe);
+                }
+                if (val != null && val.length() > 0){
+                    user = val;
+                } else {
+                    user = defuser;
+                }
+            }
+        } catch (IOException ioe) {
+            logger.log(Level.WARNING, "Error reading input", ioe);
```

Comment:
There is already a try-catch wrapping `cons.readLine()`. Is this one necessary?

Reply:
That was a refactor I missed, no it's not necessary as the log messages are the same - I originally had them different (_error instantiating_ and _error reading_).